sklearn/utils/multiclass.py

```diff
@@ -350,6 +350,22 @@ def _check_partial_fit_first_call(clf, classes=None):
     return False
 
 
+def _validate_y_against_classes(y, classes):
```

Comment:
> Is there a reason this shouldn't just be incorporated into `_check_partial_fit_first_call`?

have refactored the same!

> And I suspect this should also handle multioutput classifiers

Done!

> Also, this function will behave very strangely if a sparse multilabel indicator matrix is passed in. 

I've tested for sparsity and raised a ValueError...